autogpt_platform/backend/backend/blocks/ayrshare/_api.py

```diff
@@ -0,0 +1,482 @@
+from __future__ import annotations
+
+import json
+import logging
+from dataclasses import dataclass
+from enum import Enum
+from typing import Any, Dict, List, Optional, Union
+
+from backend.util.request import Requests
+from backend.util.settings import Settings
+
+logger = logging.getLogger(__name__)
+
+settings = Settings()
+
+
+class AyrshareAPIException(Exception):
+    def __init__(self, message: str, status_code: int):
+        super().__init__(message)
+        self.status_code = status_code
+
+
+class SocialPlatform(str, Enum):
+    BLUESKY = "bluesky"
+    FACEBOOK = "facebook"
+    TWITTER = "twitter"
+    LINKEDIN = "linkedin"
+    INSTAGRAM = "instagram"
+    YOUTUBE = "youtube"
+    REDDIT = "reddit"
+    TELEGRAM = "telegram"
+    GMB = "gmb"
+    PINTEREST = "pinterest"
+    TIKTOK = "tiktok"
+
+
+@dataclass
+class EmailConfig:
+    to: str
+    subject: Optional[str] = None
+    body: Optional[str] = None
+    from_name: Optional[str] = None
+    from_email: Optional[str] = None
+
+
+@dataclass
+class JWTResponse:
+    status: str
+    title: str
+    token: str
+    url: str
+    emailSent: Optional[bool] = None
+    expiresIn: Optional[str] = None
+
+
+@dataclass
+class ProfileResponse:
+    status: str
+    title: str
+    refId: str
+    profileKey: str
+    messagingActive: Optional[bool] = None
+
+
+@dataclass
+class PostResponse:
+    status: str
+    id: str
+    refId: str
+    profileTitle: str
+    post: str
+    postIds: Optional[List[Dict[str, Any]]] = None
+    scheduleDate: Optional[str] = None
+    errors: Optional[List[str]] = None
+
+
+@dataclass
+class AutoHashtag:
+    max: Optional[int] = None
+    position: Optional[str] = None
+
+
+@dataclass
+class FirstComment:
+    text: str
+    platforms: Optional[List[SocialPlatform]] = None
+
+
+@dataclass
+class AutoSchedule:
+    interval: str
+    platforms: Optional[List[SocialPlatform]] = None
+    startDate: Optional[str] = None
+    endDate: Optional[str] = None
+
+
+@dataclass
+class AutoRepost:
+    interval: str
+    platforms: Optional[List[SocialPlatform]] = None
+    startDate: Optional[str] = None
+    endDate: Optional[str] = None
+
+
+@dataclass
+class PostError:
+    code: int
+    message: str
+    details: str
+
+
+class AyrshareClient:
+    """Client for the Ayrshare Social Media Post API"""
+
+    API_URL = "https://api.ayrshare.com/api"
+    POST_ENDPOINT = f"{API_URL}/post"
+    PROFILES_ENDPOINT = f"{API_URL}/profiles"
+    JWT_ENDPOINT = f"{PROFILES_ENDPOINT}/generateJWT"
+
+    def __init__(
+        self,
+        custom_requests: Optional[Requests] = None,
+    ):
+        headers: Dict[str, str] = {
+            "Content-Type": "application/json",
+            "Authorization": f"Bearer {settings.secrets.ayrshare_api_key}",
```

Comment:
This should probably raise a `MissingConfigError` if Ayrshare API key isn't set
https://github.com/Significant-Gravitas/AutoGPT/blob/0739be2d9f09b9e31a91d2fd2d0563ab9f2765f0/autogpt_platform/backend/backend/util/exceptions.py#L1-L2